cpp/src/io/orc/orc_reader.cu

```diff
@@ -711,8 +682,10 @@ gdf_error read_orc(orc_read_arg *args) {
           len += stream_info[stream_count].length;
           stream_count++;
         }
-        CUDA_TRY(cudaMemcpyAsync(d_dst, input.data() + offset, len,
+        const auto buffer = input.get_buffer(offset, len);
+        CUDA_TRY(cudaMemcpyAsync(d_dst, buffer->data(), len,
                                  cudaMemcpyHostToDevice));
+        CUDA_TRY(cudaStreamSynchronize(0));
```

Comment:
I don't think this change is mentioned in the description

Reply:
Added. Right now, it's to ensure it matches Parquet with BlazingDB's change in that PR. May not be needed depending on how the Arrow zero-copy is implemented but we'll have to revisit.